lib/lg-tv-database.js

```diff
@@ -4,68 +4,112 @@
 'use strict';
 
 const {Database} = require('gateway-addon');
+const fs = require('fs');
+const mkdirp = require('mkdirp');
+const os = require('os');
+const path = require('path');
+const storage = require('node-persist');
+
+function getDataPath() {
+  let profileDir;
+  if (process.env.hasOwnProperty('MOZIOT_HOME')) {
+    profileDir = process.env.MOZIOT_HOME;
+  } else {
+    profileDir = path.join(os.homedir(), '.mozilla-iot');
+  }
+
+  return path.join(profileDir, 'data', 'lg-tv-adapter');
+}
```

Comment:
Should this test to see if userProfile is present, and if so use that? And if there is no userProfile (because the gateway is too old) then do the manual calculations.

Reply:
Yes, that's probably a good idea. I've just been avoiding doing that, since nothing at all actually supports it yet.

Reply:
Also, the [data directory PR](https://github.com/mozilla-iot/gateway/pull/2169) isn't even merged yet 😃

Reply:
I was also thinking that it would be useful to have this in a common function someplace, but it winds up being the same problem becuase if the gateway is too old, then the common function wouldn't exist and you'd need to what you're doing anyways.

Reply:
Yes, exactly.